interfaces/IDeviceInfo.h

```diff
@@ -25,136 +25,271 @@
 
 namespace Thunder {
 namespace Exchange {
-
+    /* @json 1.0.0 */
     struct EXTERNAL IDeviceInfo : virtual public Core::IUnknown {
```

Comment:
<msuybr good to keep "the old" interface IDeviceInfo as as and than just add new interfaces, like:

struct IDeviceMetadata : virtual Core::IUnknown { 
    struct Device { ..} 
};

struct IImageMetadata : virtual Core::IUnknown { 
    struct Firmware { .. }
};

struct ISystemMetadata : virtual Core::IUnknown {
   struct System { .. }
}; 

struct ISocketMetadata : virtual Core::IUnknown {
    struct Socket { ... };
};

Reply:
done